src/generation.js

```diff
@@ -207,14 +207,39 @@ class NoRepeatNGramLogitsProcessor extends LogitsProcessor {
     );
     for (const [hypo, bannedTokens] of bannedBatchTokens.entries()) {
       for (const token of bannedTokens) {
-        console.log("banned token", token);
         logits.data[hypo * logits.dims[1] + token] = -Infinity;``
       }
     }
     return logits;
   }
 }
 
+
+class RepetitionPenaltyLogitsProcessor extends LogitsProcessor {
+  constructor(penalty) {
+    super();
+    this.penalty = penalty;
+  }
+
+  _call(input_ids, logits) {
+      const numBatches = logits.dims[0]
+      const batchSize = logits.dims[1]
+      for (let batch = 0; batch < numBatches; ++batch) {
```

Comment:
I'll double check my original implementation, but if I remember correctly, you shouldn't need to iterate over batches (this is is handled by the LogitsProcessorList, which, when called, does the batching there.

This is because JS doesn't allow vectorized operations.